upstairs/src/backpressure.rs

```diff
@@ -130,36 +222,55 @@ impl BackpressureConfig {
 mod test {
     use super::*;
 
+    /// Confirm that replaying the max number of jobs / bytes will not saturate
+    #[test]
+    fn check_outstanding_backpressure() {
+        let cfg = BackpressureConfig::default();
+        let BackpressureAmount::Duration(_) =
+            cfg.get_backpressure(IO_OUTSTANDING_MAX_BYTES, 0)
+        else {
+            panic!("backpressure saturated")
+        };
+
+        let BackpressureAmount::Duration(_) =
+            cfg.get_backpressure(0, IO_OUTSTANDING_MAX_JOBS as u64)
+        else {
+            panic!("backpressure saturated")
+        };
+    }
+
     #[test]
     fn check_max_backpressure() {
         let cfg = BackpressureConfig::default();
-        let t = cfg.get_backpressure_us(
-            IO_OUTSTANDING_MAX_BYTES * 2 - 1024u64.pow(2),
-            0,
-        );
-        let timeout = Duration::from_micros(t);
+        let BackpressureAmount::Duration(timeout) = cfg
+            .get_backpressure(IO_OUTSTANDING_MAX_BYTES * 2 - 1024u64.pow(2), 0)
+        else {
+            panic!("backpressure saturated")
+        };
         println!(
             "max byte-based delay: {}",
             humantime::format_duration(timeout)
         );
         assert!(
-            timeout > Duration::from_secs(60 * 60),
+            timeout >= Duration::from_secs(30),
             "max byte-based backpressure delay is too low;
-            expected > 1 hr, got {}",
+            expected > 30 secs, got {}",
             humantime::format_duration(timeout)
         );
 
-        let t =
-            cfg.get_backpressure_us(0, IO_OUTSTANDING_MAX_JOBS as u64 * 2 - 1);
-        let timeout = Duration::from_micros(t);
+        let BackpressureAmount::Duration(timeout) =
+            cfg.get_backpressure(0, IO_OUTSTANDING_MAX_JOBS as u64 * 2 - 1)
+        else {
+            panic!("backpressure saturated")
+        };
         println!(
             "max job-based delay: {}",
             humantime::format_duration(timeout)
         );
         assert!(
-            timeout > Duration::from_secs(60 * 60),
+            timeout >= Duration::from_secs(30),
             "max job-based backpressure delay is too low;
-            expected > 1 hr, got {}",
+            expected > 30 secs, got {}",
             humantime::format_duration(timeout)
         );
```

Comment:
for completion i think we should have an explicit test below here that we do in fact get Saturated at the max value

Reply:
Done in dc3130f1bcea